src/main/java/org/candlepin/subscriptions/db/EventRecordRepository.java

```diff
@@ -52,4 +53,23 @@
      */
     Stream<EventRecord> findByAccountNumberAndTimestampGreaterThanEqualAndTimestampLessThanOrderByTimestamp(
         String accountNumber, OffsetDateTime begin, OffsetDateTime end);
+
+    /**
+     * Fetch a stream of events for a given account, event type and event source for a given time range.
+     *
+     * The events returned include those at begin and up to (but not including) end.
+     *
+     * NOTE: this query does not use `between` since between semantics are inclusive. e.g. `select * from
+     * events where timestamp between '2021-01-01T00:00:00Z' and '2021-01-01T01:00:00Z` would match events
+     * at midnight UTC and 1am UTC.
+     *
+     * @param accountNumber account number
+     * @param eventSource event source
+     * @param eventType event type
+     * @param begin start of the time range (inclusive)
+     * @param end end of the time range (exclusive)
+     * @return Stream of EventRecords
+     */
+    Stream<EventRecord> findByAccountNumberAndEventSourceAndEventTypeAndTimestampGreaterThanEqualAndTimestampLessThanOrderByTimestamp(
```

Comment:
I can't wait until we roll out JPA specifications/criteria to the whole project :D